src/routes/listing.ts

```diff
@@ -75,9 +102,16 @@ router.get("/videos/popular", async (req, res, next) => {
         path: "$channel",
         preserveNullAndEmptyArrays: true
       })
-      .project(projection(keysOf<Browsable>()));
+      .project(projection(keysOf<Browsable>()))
+      .limit(limit);
+
+    const response = {
+      content: videos,
+      cursor: videos[videos.length - 1]._id || CURSOR_START,
```

Comment:
This potentially can throw a null dereference exception if `videos.length === 0`, so you should add the optional chaining operator here.